### PR TITLE
Add "More links" addon

### LINKED ISSUE #71: [New addon] Make all links clickable on Scratch?
**Describe the new addon**

Make all links clickable even if they're not inside `scratch.mit.edu`

**Link to the new addon**

Inspired by https://github.com/Hans5958/userscripts/blob/master/Scratch/More%20Links%20in%20Scratch.user.js

### PR DESCRIPTION
**Pull request by Hans5958**
**Resolves**

This PR resolves #71.

**Changes**

This adds "More links" as a new addon.

**Reason for changes**

Why not?

**Tests**

It works on Firefox, probably will work on chrome.

### CODE REVIEW

**Comment by Hans5958** [comment]
Tip: Consider using [Restyled](https://restyled.io/)

File: addons/more-links/addon.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      "url": "userscript.js",
+      "matches": [
+        "https://scratch.mit.edu/projects/*",
+        "https://scratch.mit.edu/studios/*",
+        "https://scratch.mit.edu/users/*"
+      ]
+    }
+  ],
+  "tags": ["community"],
+  "enabled_by_default": true
```
Comment (apple502j) [inline addons/more-links/addon.json:15]: This should not be enabled by default.

File: addons/more-links/userscript.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  switch (document.location.pathname.substr(1).split("/")[0]) {
+    case "users":
+      linkifyElement(document.querySelector("#user-details"));
+      break;
+    case "projects":
+      // Need to convert #[numbers] to solve conflict between tags and external Scratch player links.
+      document.querySelectorAll(".project-description a").forEach((element) => {
+        if (/\d+/.test(element.innerHTML)) element.outerHTML = element.innerHTML;
+      });
+      linkifyElement(document.querySelector(".project-notes"));
```
Comment (apple502j) [inline addons/more-links/userscript.js:14]: Do the stuff here too

**Review by apple502j — CHANGES_REQUESTED** [review]

File: addons/more-links/addon.json (hunk 1 above)
Comment (WorldLanguages) [reply addons/more-links/addon.json:15]: Eh - debatable. But I agree, I don't think it's a good idea to enable so many addons to previous Scratch Messaging Extension users, specially considering that extension currently makes no changes to the Scratch website.

File: addons/more-links/userscript.js (hunk 2 above)
Comment (Hans5958) [reply addons/more-links/userscript.js:14]: What stuff exactly?

Comment (apple502j) [reply addons/more-links/userscript.js:14]: Hashtag conversion

**Comment by apple502j** [comment]
Project Notes and Project Descriptions behave the same - why different code?

**Comment by Hans5958** [comment]
It doesn't. The one removes links from tags with numbers for external players like forkphorus, and the other one is the main part, which converts text to links.

The different CSS selector is an error. I have fixed it accordingly.

**Comment by Hans5958** [comment]
Added support for comments and avoided editable elements.
pls review @WorldLanguages @apple502j

File: addons/more-links/userscript.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import "/libraries/linkify.min.js";
+import "/libraries/linkify-element.min.js";
+
+export default async function ({ addon }) {
+  const pageType = document.location.pathname.substr(1).split("/")[0];
+  let comments;
+
+  switch (pageType) {
+    case "users":
+      linkifyElement(document.querySelectorAll("#user-details .read-only"));
```
Comment (WorldLanguages) [inline addons/more-links/userscript.js:10]: This will throw, you can't pass a node list to linkify.

Comment (Hans5958) [reply addons/more-links/userscript.js:10]: No, it isn't.

Comment (WorldLanguages) [reply addons/more-links/userscript.js:10]: I tried it - it threw in exactly that line :(

Comment (WorldLanguages) [reply addons/more-links/userscript.js:10]: image

Comment (Hans5958) [reply addons/more-links/userscript.js:10]: Nvm, it is. Fixed in the next commit.

**Review by WorldLanguages — APPROVED** [review]

**Comment by Hans5958** [comment]
Dismiss @apple502j stale review?

**Comment by WorldLanguages** [comment]
lol be patient I push slowly to enjoy the process :P

**Pull request merged**